tsinfer/eval_util.py

```diff
@@ -716,10 +716,138 @@ def count_sample_child_edges(ts):
     """
     child_counts = np.bincount(ts.tables.edges.child)
     return child_counts[ts.samples()]
-# def mean_sample_ancestry(ts):
-#     """
-#     Computes the mean sample ancestry for each node in the tree sequence. This is
-#     defined as the fraction of samples below a given node from each population,
-#     averaged along the length of sequence that the node is ancestral to at
-#     least one sample.
-#     """
+
+
+def node_span(ts):
+    """
+    Returns the "span" of all nodes in the tree sequence. This is defined as the
+    total length of all trees that the node participates in. A node "participates in"
```

Comment:
I usually think of "total length" of a tree as being the sum of all the branch lengths (times the distance along the sequence it extends for, maybe); but here you just mean the distance along the sequence.